dat does not build because of broken sodium-native dependency

**Describe the bug**
A clear and concise description of what the bug is.

Installing `nodePackages.dat` gives:
```

> utp-native@1.7.3 install /nix/store/509i6ksf5wcpibrxmgvn3jfv3fm2g87d-node_dat-13.13.1/lib/node_modules/dat/node_modules/discovery-swarm/node_modules/utp-native
> node-gyp-build

make: Entering directory '/nix/store/509i6ksf5wcpibrxmgvn3jfv3fm2g87d-node_dat-13.13.1/lib/node_modules/dat/node_modules/discovery-swarm/node_modules/utp-native/build'
  CXX(target) Release/obj.target/libutp/deps/libutp/utp_internal.o
../deps/libutp/utp_internal.cpp: In function 'size_t utp_process_incoming(UTPSocket*, const byte*, size_t, bool)':
../deps/libutp/utp_internal.cpp:1970:34: warning: comparison of integer expressions of different signedness: 'int' and 'uint32' {aka 'unsigned int'} [-Wsign-compare]
   if (conn->mtu_probe_seq && seq == conn->mtu_probe_seq) {
                              ~~~~^~~~~~~~~~~~~~~~~~~~~~
  CXX(target) Release/obj.target/libutp/deps/libutp/utp_utils.o
  CXX(target) Release/obj.target/libutp/deps/libutp/utp_hash.o
  CXX(target) Release/obj.target/libutp/deps/libutp/utp_callbacks.o
  CXX(target) Release/obj.target/libutp/deps/libutp/utp_api.o
  CXX(target) Release/obj.target/libutp/deps/libutp/utp_packedsockaddr.o
  AR(target) Release/obj.target/deps/utp.a
  COPY Release/utp.a
  CXX(target) Release/obj.target/utp/src/utp_uv.o
In file included from ../src/utp_uv.cc:2:
../../../../nan/nan.h: In function 'void Nan::AsyncQueueWorker(Nan::AsyncWorker*)':
../../../../nan/nan.h:2298:62: warning: cast between incompatible function types from 'void (*)(uv_work_t*)' {aka 'void (*)(uv_work_s*)'} to 'uv_after_work_cb' {aka 'void (*)(uv_work_s*, int)'} [-Wcast-function-type]
     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
                                                              ^
In file included from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node.h:63,
                 from ../../../../nan/nan.h:54,
                 from ../src/utp_uv.cc:2:
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h: In instantiation of 'void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = node::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)]':
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node_object_wrap.h:84:78:   required from here
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:9502:16: warning: cast between incompatible function types from 'v8::WeakCallbackInfo<node::ObjectWrap>::Callback' {aka 'void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)'} to 'Callback' {aka 'void (*)(const v8::WeakCallbackInfo<void>&)'} [-Wcast-function-type]
                reinterpret_cast<Callback>(callback), type);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h: In instantiation of 'void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = Nan::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)]':
../../../../nan/nan_object_wrap.h:65:61:   required from here
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:9502:16: warning: cast between incompatible function types from 'v8::WeakCallbackInfo<Nan::ObjectWrap>::Callback' {aka 'void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)'} to 'Callback' {aka 'void (*)(const v8::WeakCallbackInfo<void>&)'} [-Wcast-function-type]
../src/utp_uv.cc: At global scope:
../src/utp_uv.cc:219:1: warning: 'uint64 on_utp_schedule_ack(utp_callback_arguments*)' defined but not used [-Wunused-function]
 on_utp_schedule_ack (utp_callback_arguments *a) {
 ^~~~~~~~~~~~~~~~~~~
  CXX(target) Release/obj.target/utp/src/socket_wrap.o
In file included from ../src/socket_wrap.h:4,
                 from ../src/socket_wrap.cc:1:
../../../../nan/nan.h: In function 'void Nan::AsyncQueueWorker(Nan::AsyncWorker*)':
../../../../nan/nan.h:2298:62: warning: cast between incompatible function types from 'void (*)(uv_work_t*)' {aka 'void (*)(uv_work_s*)'} to 'uv_after_work_cb' {aka 'void (*)(uv_work_s*, int)'} [-Wcast-function-type]
     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
                                                              ^
../src/socket_wrap.cc: In member function 'int SocketWrap::Drain()':
../src/socket_wrap.cc:69:40: warning: 'v8::Local<v8::Value> Nan::Callback::Call(v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const' is deprecated [-Wdeprecated-declarations]
       this->on_drain->Call(ctx, 0, NULL);
                                        ^
In file included from ../src/socket_wrap.h:4,
                 from ../src/socket_wrap.cc:1:
../../../../nan/nan.h:1718:3: note: declared here
   Call(v8::Local<v8::Object> target
   ^~~~
../src/socket_wrap.cc: In static member function 'static Nan::NAN_METHOD_RETURN_TYPE SocketWrap::Context(Nan::NAN_METHOD_ARGS_TYPE)':
../src/socket_wrap.cc:99:45: warning: 'v8::Local<v8::Object> v8::Value::ToObject() const' is deprecated: Use maybe version [-Wdeprecated-declarations]
   Local<Object> context = info[0]->ToObject();
                                             ^
In file included from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node.h:63,
                 from ../../../../nan/nan.h:54,
                 from ../src/socket_wrap.h:4,
                 from ../src/socket_wrap.cc:1:
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:10046:15: note: declared here
 Local<Object> Value::ToObject() const {
               ^~~~~
../src/socket_wrap.cc: In static member function 'static Nan::NAN_METHOD_RETURN_TYPE SocketWrap::Write(Nan::NAN_METHOD_ARGS_TYPE)':
../src/socket_wrap.cc:113:43: warning: 'v8::Local<v8::Object> v8::Value::ToObject() const' is deprecated: Use maybe version [-Wdeprecated-declarations]
   Local<Value> buffer = info[0]->ToObject();
                                           ^
In file included from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node.h:63,
                 from ../../../../nan/nan.h:54,
                 from ../src/socket_wrap.h:4,
                 from ../src/socket_wrap.cc:1:
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:10046:15: note: declared here
 Local<Object> Value::ToObject() const {
               ^~~~~
../src/socket_wrap.cc: In static member function 'static Nan::NAN_METHOD_RETURN_TYPE SocketWrap::Writev(Nan::NAN_METHOD_ARGS_TYPE)':
../src/socket_wrap.cc:141:61: warning: 'v8::Local<v8::Object> v8::Value::ToObject() const' is deprecated: Use maybe version [-Wdeprecated-declarations]
     Local<Value> buffer = Nan::Get(writes->Get(i)->ToObject(), chunk).ToLocalChecked();
                                                             ^
In file included from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node.h:63,
                 from ../../../../nan/nan.h:54,
                 from ../src/socket_wrap.h:4,
                 from ../src/socket_wrap.cc:1:
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:10046:15: note: declared here
 Local<Object> Value::ToObject() const {
               ^~~~~
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h: In instantiation of 'void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = node::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)]':
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node_object_wrap.h:84:78:   required from here
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:9502:16: warning: cast between incompatible function types from 'v8::WeakCallbackInfo<node::ObjectWrap>::Callback' {aka 'void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)'} to 'Callback' {aka 'void (*)(const v8::WeakCallbackInfo<void>&)'} [-Wcast-function-type]
                reinterpret_cast<Callback>(callback), type);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h: In instantiation of 'void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = Nan::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)]':
../../../../nan/nan_object_wrap.h:65:61:   required from here
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:9502:16: warning: cast between incompatible function types from 'v8::WeakCallbackInfo<Nan::ObjectWrap>::Callback' {aka 'void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)'} to 'Callback' {aka 'void (*)(const v8::WeakCallbackInfo<void>&)'} [-Wcast-function-type]
  CXX(target) Release/obj.target/utp/src/utp_wrap.o
In file included from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
../../../../nan/nan.h: In function 'void Nan::AsyncQueueWorker(Nan::AsyncWorker*)':
../../../../nan/nan.h:2298:62: warning: cast between incompatible function types from 'void (*)(uv_work_t*)' {aka 'void (*)(uv_work_s*)'} to 'uv_after_work_cb' {aka 'void (*)(uv_work_s*, int)'} [-Wcast-function-type]
     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
                                                              ^
../src/utp_wrap.cc: In function 'void callback_on_message(utp_uv_t*, char*, size_t, int, char*)':
../src/utp_wrap.cc:25:38: warning: 'v8::Local<v8::Value> Nan::Callback::Call(v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const' is deprecated [-Wdeprecated-declarations]
   wrap->on_message->Call(ctx, 2, argv);
                                      ^
In file included from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
../../../../nan/nan.h:1718:3: note: declared here
   Call(v8::Local<v8::Object> target
   ^~~~
../src/utp_wrap.cc: In function 'void callback_on_send(utp_uv_t*, uv_udp_send_t*, int)':
../src/utp_wrap.cc:39:35: warning: 'v8::Local<v8::Value> Nan::Callback::Call(v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const' is deprecated [-Wdeprecated-declarations]
   wrap->on_send->Call(ctx, 2, argv);
                                   ^
In file included from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
../../../../nan/nan.h:1718:3: note: declared here
   Call(v8::Local<v8::Object> target
   ^~~~
../src/utp_wrap.cc: In function 'void callback_on_close(utp_uv_t*)':
../src/utp_wrap.cc:49:36: warning: 'v8::Local<v8::Value> Nan::Callback::Call(v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const' is deprecated [-Wdeprecated-declarations]
   wrap->on_close->Call(ctx, 0, NULL);
                                    ^
In file included from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
../../../../nan/nan.h:1718:3: note: declared here
   Call(v8::Local<v8::Object> target
   ^~~~
../src/utp_wrap.cc: In function 'void callback_on_error(utp_uv_t*)':
../src/utp_wrap.cc:59:36: warning: 'v8::Local<v8::Value> Nan::Callback::Call(v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const' is deprecated [-Wdeprecated-declarations]
   wrap->on_error->Call(ctx, 0, NULL);
                                    ^
In file included from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
../../../../nan/nan.h:1718:3: note: declared here
   Call(v8::Local<v8::Object> target
   ^~~~
../src/utp_wrap.cc: In function 'void callback_on_socket_read(utp_uv_t*, utp_socket*, char*, size_t)':
../src/utp_wrap.cc:72:35: warning: 'v8::Local<v8::Value> Nan::Callback::Call(v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const' is deprecated [-Wdeprecated-declarations]
   wrap->on_data->Call(ctx, 1, argv);
                                   ^
In file included from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
../../../../nan/nan.h:1718:3: note: declared here
   Call(v8::Local<v8::Object> target
   ^~~~
../src/utp_wrap.cc: In function 'void callback_on_socket_end(utp_uv_t*, utp_socket*)':
../src/utp_wrap.cc:86:34: warning: 'v8::Local<v8::Value> Nan::Callback::Call(v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const' is deprecated [-Wdeprecated-declarations]
   wrap->on_end->Call(ctx, 0, NULL);
                                  ^
In file included from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
../../../../nan/nan.h:1718:3: note: declared here
   Call(v8::Local<v8::Object> target
   ^~~~
../src/utp_wrap.cc: In function 'void callback_on_socket_error(utp_uv_t*, utp_socket*, int)':
../src/utp_wrap.cc:107:36: warning: 'v8::Local<v8::Value> Nan::Callback::Call(v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const' is deprecated [-Wdeprecated-declarations]
   wrap->on_error->Call(ctx, 1, argv);
                                    ^
In file included from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
../../../../nan/nan.h:1718:3: note: declared here
   Call(v8::Local<v8::Object> target
   ^~~~
../src/utp_wrap.cc: In function 'void callback_on_socket_close(utp_uv_t*, utp_socket*)':
../src/utp_wrap.cc:117:36: warning: 'v8::Local<v8::Value> Nan::Callback::Call(v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const' is deprecated [-Wdeprecated-declarations]
   wrap->on_close->Call(ctx, 0, NULL);
                                    ^
In file included from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
../../../../nan/nan.h:1718:3: note: declared here
   Call(v8::Local<v8::Object> target
   ^~~~
../src/utp_wrap.cc: In function 'void callback_on_socket_connect(utp_uv_t*, utp_socket*)':
../src/utp_wrap.cc:129:38: warning: 'v8::Local<v8::Value> Nan::Callback::Call(v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const' is deprecated [-Wdeprecated-declarations]
   wrap->on_connect->Call(ctx, 0, NULL);
                                      ^
In file included from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
../../../../nan/nan.h:1718:3: note: declared here
   Call(v8::Local<v8::Object> target
   ^~~~
../src/utp_wrap.cc: In function 'void callback_on_socket(utp_uv_t*, utp_socket*)':
../src/utp_wrap.cc:144:89: warning: 'v8::Local<v8::Object> v8::Value::ToObject() const' is deprecated: Use maybe version [-Wdeprecated-declarations]
   SocketWrap *socket_unwrap = Nan::ObjectWrap::Unwrap<SocketWrap>(socket_wrap->ToObject());
                                                                                         ^
In file included from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node.h:63,
                 from ../../../../nan/nan.h:54,
                 from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:10046:15: note: declared here
 Local<Object> Value::ToObject() const {
               ^~~~~
../src/utp_wrap.cc:149:37: warning: 'v8::Local<v8::Value> Nan::Callback::Call(v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const' is deprecated [-Wdeprecated-declarations]
   wrap->on_socket->Call(ctx, 1, argv);
                                     ^
In file included from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
../../../../nan/nan.h:1718:3: note: declared here
   Call(v8::Local<v8::Object> target
   ^~~~
../src/utp_wrap.cc: In static member function 'static Nan::NAN_METHOD_RETURN_TYPE UTPWrap::Bind(Nan::NAN_METHOD_ARGS_TYPE)':
../src/utp_wrap.cc:205:35: warning: 'uint32_t v8::Value::Uint32Value() const' is deprecated: Use maybe version [-Wdeprecated-declarations]
   int port = info[0]->Uint32Value();
                                   ^
In file included from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:26,
                 from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node.h:63,
                 from ../../../../nan/nan.h:54,
                 from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:2477:47: note: declared here
   V8_DEPRECATED("Use maybe version", uint32_t Uint32Value() const);
                                               ^~~~~~~~~~~
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8config.h:324:3: note: in definition of macro 'V8_DEPRECATED'
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/utp_wrap.cc: In static member function 'static Nan::NAN_METHOD_RETURN_TYPE UTPWrap::Context(Nan::NAN_METHOD_ARGS_TYPE)':
../src/utp_wrap.cc:218:45: warning: 'v8::Local<v8::Object> v8::Value::ToObject() const' is deprecated: Use maybe version [-Wdeprecated-declarations]
   Local<Object> context = info[0]->ToObject();
                                             ^
In file included from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node.h:63,
                 from ../../../../nan/nan.h:54,
                 from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:10046:15: note: declared here
 Local<Object> Value::ToObject() const {
               ^~~~~
../src/utp_wrap.cc: In static member function 'static Nan::NAN_METHOD_RETURN_TYPE UTPWrap::MaxSockets(Nan::NAN_METHOD_ARGS_TYPE)':
../src/utp_wrap.cc:250:55: warning: 'uint32_t v8::Value::Uint32Value() const' is deprecated: Use maybe version [-Wdeprecated-declarations]
   handle->max_sockets = (size_t) info[0]->Uint32Value();
                                                       ^
In file included from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:26,
                 from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node.h:63,
                 from ../../../../nan/nan.h:54,
                 from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:2477:47: note: declared here
   V8_DEPRECATED("Use maybe version", uint32_t Uint32Value() const);
                                               ^~~~~~~~~~~
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8config.h:324:3: note: in definition of macro 'V8_DEPRECATED'
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/utp_wrap.cc: In static member function 'static Nan::NAN_METHOD_RETURN_TYPE UTPWrap::Send(Nan::NAN_METHOD_ARGS_TYPE)':
../src/utp_wrap.cc:257:45: warning: 'uint32_t v8::Value::Uint32Value() const' is deprecated: Use maybe version [-Wdeprecated-declarations]
   size_t id = (size_t) info[0]->Uint32Value();
                                             ^
In file included from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:26,
                 from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node.h:63,
                 from ../../../../nan/nan.h:54,
                 from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:2477:47: note: declared here
   V8_DEPRECATED("Use maybe version", uint32_t Uint32Value() const);
                                               ^~~~~~~~~~~
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8config.h:324:3: note: in definition of macro 'V8_DEPRECATED'
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/utp_wrap.cc:259:44: warning: 'v8::Local<v8::Object> v8::Value::ToObject() const' is deprecated: Use maybe version [-Wdeprecated-declarations]
   Local<Object> buffer = info[1]->ToObject();
                                            ^
In file included from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node.h:63,
                 from ../../../../nan/nan.h:54,
                 from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:10046:15: note: declared here
 Local<Object> Value::ToObject() const {
               ^~~~~
../src/utp_wrap.cc:260:37: warning: 'uint32_t v8::Value::Uint32Value() const' is deprecated: Use maybe version [-Wdeprecated-declarations]
   int offset = info[2]->Uint32Value();
                                     ^
In file included from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:26,
                 from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node.h:63,
                 from ../../../../nan/nan.h:54,
                 from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:2477:47: note: declared here
   V8_DEPRECATED("Use maybe version", uint32_t Uint32Value() const);
                                               ^~~~~~~~~~~
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8config.h:324:3: note: in definition of macro 'V8_DEPRECATED'
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/utp_wrap.cc:261:34: warning: 'uint32_t v8::Value::Uint32Value() const' is deprecated: Use maybe version [-Wdeprecated-declarations]
   int len = info[3]->Uint32Value();
                                  ^
In file included from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:26,
                 from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node.h:63,
                 from ../../../../nan/nan.h:54,
                 from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:2477:47: note: declared here
   V8_DEPRECATED("Use maybe version", uint32_t Uint32Value() const);
                                               ^~~~~~~~~~~
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8config.h:324:3: note: in definition of macro 'V8_DEPRECATED'
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/utp_wrap.cc:262:35: warning: 'uint32_t v8::Value::Uint32Value() const' is deprecated: Use maybe version [-Wdeprecated-declarations]
   int port = info[4]->Uint32Value();
                                   ^
In file included from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:26,
                 from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node.h:63,
                 from ../../../../nan/nan.h:54,
                 from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:2477:47: note: declared here
   V8_DEPRECATED("Use maybe version", uint32_t Uint32Value() const);
                                               ^~~~~~~~~~~
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8config.h:324:3: note: in definition of macro 'V8_DEPRECATED'
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/utp_wrap.cc: In static member function 'static Nan::NAN_METHOD_RETURN_TYPE UTPWrap::Connect(Nan::NAN_METHOD_ARGS_TYPE)':
../src/utp_wrap.cc:284:35: warning: 'uint32_t v8::Value::Uint32Value() const' is deprecated: Use maybe version [-Wdeprecated-declarations]
   int port = info[0]->Uint32Value();
                                   ^
In file included from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:26,
                 from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node.h:63,
                 from ../../../../nan/nan.h:54,
                 from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:2477:47: note: declared here
   V8_DEPRECATED("Use maybe version", uint32_t Uint32Value() const);
                                               ^~~~~~~~~~~
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8config.h:324:3: note: in definition of macro 'V8_DEPRECATED'
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/utp_wrap.cc:296:89: warning: 'v8::Local<v8::Object> v8::Value::ToObject() const' is deprecated: Use maybe version [-Wdeprecated-declarations]
   SocketWrap *socket_unwrap = Nan::ObjectWrap::Unwrap<SocketWrap>(socket_wrap->ToObject());
                                                                                         ^
In file included from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node.h:63,
                 from ../../../../nan/nan.h:54,
                 from ../src/utp_wrap.h:4,
                 from ../src/utp_wrap.cc:1:
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:10046:15: note: declared here
 Local<Object> Value::ToObject() const {
               ^~~~~
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h: In instantiation of 'void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = node::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)]':
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node_object_wrap.h:84:78:   required from here
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:9502:16: warning: cast between incompatible function types from 'v8::WeakCallbackInfo<node::ObjectWrap>::Callback' {aka 'void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)'} to 'Callback' {aka 'void (*)(const v8::WeakCallbackInfo<void>&)'} [-Wcast-function-type]
                reinterpret_cast<Callback>(callback), type);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h: In instantiation of 'void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = Nan::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)]':
../../../../nan/nan_object_wrap.h:65:61:   required from here
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:9502:16: warning: cast between incompatible function types from 'v8::WeakCallbackInfo<Nan::ObjectWrap>::Callback' {aka 'void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)'} to 'Callback' {aka 'void (*)(const v8::WeakCallbackInfo<void>&)'} [-Wcast-function-type]
  CXX(target) Release/obj.target/utp/binding.o
In file included from ../binding.cc:1:
../../../../nan/nan.h: In function 'void Nan::AsyncQueueWorker(Nan::AsyncWorker*)':
../../../../nan/nan.h:2298:62: warning: cast between incompatible function types from 'void (*)(uv_work_t*)' {aka 'void (*)(uv_work_s*)'} to 'uv_after_work_cb' {aka 'void (*)(uv_work_s*, int)'} [-Wcast-function-type]
     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
                                                              ^
In file included from ../../../../nan/nan.h:54,
                 from ../binding.cc:1:
../binding.cc: At global scope:
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node.h:573:43: warning: cast between incompatible function types from 'void (*)(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)' {aka 'void (*)(v8::Local<v8::Object>)'} to 'node::addon_register_func' {aka 'void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)'} [-Wcast-function-type]
       (node::addon_register_func) (regfunc),                          \
                                           ^
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node.h:607:3: note: in expansion of macro 'NODE_MODULE_X'
   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
   ^~~~~~~~~~~~~
../binding.cc:15:1: note: in expansion of macro 'NODE_MODULE'
 NODE_MODULE(utp, InitAll)
 ^~~~~~~~~~~
In file included from /nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node.h:63,
                 from ../../../../nan/nan.h:54,
                 from ../binding.cc:1:
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h: In instantiation of 'void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = node::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)]':
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/src/node_object_wrap.h:84:78:   required from here
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:9502:16: warning: cast between incompatible function types from 'v8::WeakCallbackInfo<node::ObjectWrap>::Callback' {aka 'void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)'} to 'Callback' {aka 'void (*)(const v8::WeakCallbackInfo<void>&)'} [-Wcast-function-type]
                reinterpret_cast<Callback>(callback), type);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h: In instantiation of 'void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = Nan::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)]':
../../../../nan/nan_object_wrap.h:65:61:   required from here
/nix/store/6j82b61p9ffvkhaad00skzq6jhb1dwdi-node-sources/deps/v8/include/v8.h:9502:16: warning: cast between incompatible function types from 'v8::WeakCallbackInfo<Nan::ObjectWrap>::Callback' {aka 'void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)'} to 'Callback' {aka 'void (*)(const v8::WeakCallbackInfo<void>&)'} [-Wcast-function-type]
  SOLINK_MODULE(target) Release/obj.target/utp.node
  COPY Release/utp.node
make: Leaving directory '/nix/store/509i6ksf5wcpibrxmgvn3jfv3fm2g87d-node_dat-13.13.1/lib/node_modules/dat/node_modules/discovery-swarm/node_modules/utp-native/build'
90m..................] / : info lifecycle utp-native@1.7.3~install: utp-native@1.
> utp-native@2.1.4 install /nix/store/509i6ksf5wcpibrxmgvn3jfv3fm2g87d-node_dat-13.13.1/lib/node_modules/dat/node_modules/utp-native
> node-gyp-build

...............] - : info lifecycle utp-native@2.1.4~install: utp-native@2.[0m
> fd-lock@1.0.2 install /nix/store/509i6ksf5wcpibrxmgvn3jfv3fm2g87d-node_dat-13.13.1/lib/node_modules/dat/node_modules/fd-lock
> node-gyp-build


> sodium-native@2.4.6 install /nix/store/509i6ksf5wcpibrxmgvn3jfv3fm2g87d-node_dat-13.13.1/lib/node_modules/dat/node_modules/sodium-native
> node-gyp-build "node preinstall.js" "node postinstall.js"

libtool is required, but wasn't found on this system
./configure: line 5: ./configure: No such file or directory
/nix/store/509i6ksf5wcpibrxmgvn3jfv3fm2g87d-node_dat-13.13.1/lib/node_modules/dat/node_modules/sodium-native/preinstall.js:119
    if (err) throw err
             ^

Error: ./configure exited with 127
    at ChildProcess.<anonymous> (/nix/store/509i6ksf5wcpibrxmgvn3jfv3fm2g87d-node_dat-13.13.1/lib/node_modules/dat/node_modules/sodium-native/preinstall.js:149:25)
    at ChildProcess.emit (events.js:198:13)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:248:12)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! sodium-native@2.4.6 install: `node-gyp-build "node preinstall.js" "node postinstall.js"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the sodium-native@2.4.6 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /build/.npm/_logs/2019-12-28T15_44_44_113Z-debug.log

builder for '/nix/store/nvrsj3fkd0qzc6rm26s2zjny0h3n5va2-node_dat-13.13.1.drv' failed with exit code 1
error: build of '/nix/store/nvrsj3fkd0qzc6rm26s2zjny0h3n5va2-node_dat-13.13.1.drv' failed

```
Note that installing dat with npm gives:
<description>

```
npm -g i dat                                                                                                                                ✔ 
/home/paki/.npm/bin/dat -> /home/paki/.npm/lib/node_modules/dat/bin/cli.js

> fd-lock@1.0.2 install /home/paki/.npm/lib/node_modules/dat/node_modules/fd-lock
> node-gyp-build


> sodium-native@2.4.6 install /home/paki/.npm/lib/node_modules/dat/node_modules/sodium-native
> node-gyp-build "node preinstall.js" "node postinstall.js"

libtool is required, but wasn't found on this system
./configure: riga 5: ./configure: No such file or directory
/home/paki/.npm/lib/node_modules/dat/node_modules/sodium-native/preinstall.js:119
    if (err) throw err
             ^

Error: ./configure exited with 127
    at ChildProcess.<anonymous> (/home/paki/.npm/lib/node_modules/dat/node_modules/sodium-native/preinstall.js:149:25)
    at ChildProcess.emit (events.js:219:5)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:274:12)

> utp-native@1.7.3 install /home/paki/.npm/lib/node_modules/dat/node_modules/discovery-swarm/node_modules/utp-native
> node-gyp-build

make: ingresso nella directory "/home/paki/.npm/lib/node_modules/dat/node_modules/discovery-swarm/node_modules/utp-native/build"
  CXX(target) Release/obj.target/libutp/deps/libutp/utp_internal.o
../deps/libutp/utp_internal.cpp: In function ‘size_t utp_process_incoming(UTPSocket*, const byte*, size_t, bool)’:
../deps/libutp/utp_internal.cpp:1970:34: warning: comparison of integer expressions of different signedness: ‘int’ and ‘uint32’ {aka ‘unsigned int’} [-Wsign-compare]
   if (conn->mtu_probe_seq && seq == conn->mtu_probe_seq) {
                              ~~~~^~~~~~~~~~~~~~~~~~~~~~
  CXX(target) Release/obj.target/libutp/deps/libutp/utp_utils.o
  CXX(target) Release/obj.target/libutp/deps/libutp/utp_hash.o
  CXX(target) Release/obj.target/libutp/deps/libutp/utp_callbacks.o
  CXX(target) Release/obj.target/libutp/deps/libutp/utp_api.o
  CXX(target) Release/obj.target/libutp/deps/libutp/utp_packedsockaddr.o
  AR(target) Release/obj.target/deps/utp.a
/bin/sh: ar: comando non trovato
make: *** [deps/libutp.target.mk:138: Release/obj.target/deps/utp.a] Error 127
make: uscita dalla directory "/home/paki/.npm/lib/node_modules/dat/node_modules/discovery-swarm/node_modules/utp-native/build"
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/nix/store/gaxkyajm03s2dd2f9nbmk5pm7iwgybg5-nodejs-13.3.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:219:5)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:274:12)
gyp ERR! System Linux 5.4.1
gyp ERR! command "/nix/store/gaxkyajm03s2dd2f9nbmk5pm7iwgybg5-nodejs-13.3.0/bin/node" "/nix/store/gaxkyajm03s2dd2f9nbmk5pm7iwgybg5-nodejs-13.3.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/paki/.npm/lib/node_modules/dat/node_modules/discovery-swarm/node_modules/utp-native
gyp ERR! node -v v13.3.0
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok 

> utp-native@2.1.5 install /home/paki/.npm/lib/node_modules/dat/node_modules/utp-native
> node-gyp-build

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: sodium-native@2.4.6 (node_modules/dat/node_modules/sodium-native):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: sodium-native@2.4.6 install: `node-gyp-build "node preinstall.js" "node postinstall.js"`
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Exit status 1
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: utp-native@1.7.3 (node_modules/dat/node_modules/discovery-swarm/node_modules/utp-native):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: utp-native@1.7.3 install: `node-gyp-build`
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Exit status 1

+ dat@13.13.1
added 467 packages from 244 contributors in 58.231s

```

</description>
So it seems that  `sodium-native`  is an optional dependency and does not seem to be needed.


**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.1, NixOS, 20.03pre206632.b0bbacb5213 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.1`
 - channels(root): `"nixos-20.03pre206632.b0bbacb5213, nur, home-manager"`
 - channels(paki): `"nixos-20.03pre196201.07d4df59626"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`



Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - nodePackages.dat
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/node-packages/default-v10.nix
+++ b/pkgs/development/node-packages/default-v10.nix
@@ -21,7 +21,7 @@ nodePackages // {
   };
 
   dat = nodePackages.dat.override {
-    buildInputs = [ nodePackages.node-gyp-build ];
+    buildInputs = [ nodePackages.node-gyp-build pkgs.libtool pkgs.autoconf pkgs.automake ];
   };
 
   dnschain = nodePackages.dnschain.override {

